services/src/api/model/db_types.rs

```diff
@@ -1539,6 +1553,540 @@ impl TryFrom<MetaDataDefinitionDbType> for MetaDataDefinition {
+    fn from(other: &EbvPortalDataProviderDefinition) -> Self {
+        Self {
+            name: other.name.clone(),
+            path: other.path.to_string_lossy().to_string(),
+            base_url: other.base_url.clone().into(),
+            overviews: other.overviews.to_string_lossy().to_string(),
+            cache_ttl: other.cache_ttl,
+        }
+    }
+}
+
+impl TryFrom<EbvPortalDataProviderDefinitionDbType> for EbvPortalDataProviderDefinition {
+    type Error = Error;
+
+    fn try_from(other: EbvPortalDataProviderDefinitionDbType) -> Result<Self, Self::Error> {
+        Ok(Self {
+            name: other.name,
+            path: other.path.into(),
+            base_url: other.base_url.as_str().try_into()?,
+            overviews: other.overviews.into(),
+            cache_ttl: other.cache_ttl,
+        })
+    }
+}
+
+#[derive(Debug, ToSql, FromSql)]
+#[postgres(name = "NetCdfCfDataProviderDefinition")]
+pub struct NetCdfCfDataProviderDefinitionDbType {
+    pub name: String,
+    pub path: String,
+    pub overviews: String,
+    pub cache_ttl: CacheTtlSeconds,
+}
+
+impl From<&NetCdfCfDataProviderDefinition> for NetCdfCfDataProviderDefinitionDbType {
+    fn from(other: &NetCdfCfDataProviderDefinition) -> Self {
+        Self {
+            name: other.name.clone(),
+            path: other.path.to_string_lossy().to_string(),
+            overviews: other.overviews.to_string_lossy().to_string(),
+            cache_ttl: other.cache_ttl,
+        }
+    }
+}
+
+impl TryFrom<NetCdfCfDataProviderDefinitionDbType> for NetCdfCfDataProviderDefinition {
+    type Error = Error;
+
+    fn try_from(other: NetCdfCfDataProviderDefinitionDbType) -> Result<Self, Self::Error> {
+        Ok(Self {
+            name: other.name,
+            path: other.path.into(),
+            overviews: other.overviews.into(),
+            cache_ttl: other.cache_ttl,
+        })
+    }
+}
+
+#[derive(Debug, ToSql, FromSql)]
+#[postgres(name = "PangaeaDataProviderDefinition")]
+pub struct PangaeaDataProviderDefinitionDbType {
+    pub name: String,
+    pub base_url: String,
+    pub cache_ttl: CacheTtlSeconds,
+}
+
+impl From<&PangaeaDataProviderDefinition> for PangaeaDataProviderDefinitionDbType {
+    fn from(other: &PangaeaDataProviderDefinition) -> Self {
+        Self {
+            name: other.name.clone(),
+            base_url: other.base_url.clone().into(),
+            cache_ttl: other.cache_ttl,
+        }
+    }
+}
+
+impl TryFrom<PangaeaDataProviderDefinitionDbType> for PangaeaDataProviderDefinition {
+    type Error = Error;
+
+    fn try_from(other: PangaeaDataProviderDefinitionDbType) -> Result<Self, Self::Error> {
+        Ok(Self {
+            name: other.name,
+            base_url: other.base_url.as_str().try_into()?,
+            cache_ttl: other.cache_ttl,
+        })
+    }
+}
+
+#[derive(Debug, ToSql, FromSql)]
+#[postgres(name = "MockLayer")]
+pub struct MockLayerDbType {
```

Comment:
do we really need to persist mock data? Do we need mock data outside of tests? If it's only for tests, maybe it's ok that they do not survive a restart of the engine?

Reply:
[afacdd0](https://github.com/geo-engine/geoengine/pull/866/commits/afacdd00f469216470891236b82bbec8bb532ea0)